test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```
@@ -0,0 +1,114 @@
+
+    private UsernamePasswordAuthenticationCredentials usernamePasswordAuthenticationCredentials;
+    private UsernamePasswordAuthenticationCredentials notConfiguredInstance;
+    private String username = "myUsername";
+    private String password = "myPassword";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+        this.notConfiguredInstance = new UsernamePasswordAuthenticationCredentials();
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithTheCorrectFieldSeparator() throws Exception {
+        // Given
+        List<String> someCorrectSeparators = Arrays.asList("-", "|", "/", "\\", "+");
+        String aCorrectSeparator = someCorrectSeparators.get(new Random().nextInt(someCorrectSeparators.size()));
```

It could test all separators, something like:
```Java
for (String correctSeparator : someCorrectSeparators) {
    // When
    String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(correctSeparator);

    // Then
    assertThat(String.format("Failed to encode with '%s'", correctSeparator), encodedUsernamePassword, notNullValue());
    assertThat(
            String.format("Failed to properly encode with '%s'", correctSeparator),
            encodedUsernamePassword,
            is("bXlVc2VybmFtZQ==~bXlQYXNzd29yZA==~"));
}
```
(In any case it should indicate which separator was used.)

Also, maybe we should throw an exception instead of silently continue the encoding when "wrong" parent separator is used?